### PR TITLE
First round of tests for the demo mobile app

### PR DESCRIPTION
**Pull request by marcelomorgado**
Refs #12 

Snapshots: https://github.com/tasitlabs/tasit/issues/12#issuecomment-448298933
Enzyme: https://github.com/tasitlabs/tasit/issues/12#issuecomment-448300733

### CODE REVIEW

File: demo/components/presentational/EthereumSignUpForm.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
 import React from "react";
 import { Button, StyleSheet, View, Text, TextInput } from "react-native";
+import { Account } from "tasit-sdk";
 
 export default class EthereumSignUpForm extends React.Component {
   state = {
     text: "",
+    address: "",
```
Comment (pcowgill) [inline demo/components/presentational/EthereumSignUpForm.js]: This is a good candidate for moving to redux in the future

File: demo/components/presentational/EthereumSignUpForm.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  it("renders the component", async () => {
+    expect(shallow(<EthereumSignUpForm />)).toMatchSnapshot();
+  });
+
+  it("creates a wallet - calling function", async () => {
+    const wrapper = shallow(<EthereumSignUpForm />);
+    expect(wrapper.state("address")).toEqual("");
+    await wrapper.instance().createAccount();
+    expect(wrapper.state("address")).not.toEqual("");
```
Comment (pcowgill) [inline demo/components/presentational/EthereumSignUpForm.test.js]: We could assert that the address is the right length here too. But maybe the Tasit SDK tests will already cover that sufficiently.

File: demo/jest.preprocessor.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+/* eslint-env node */
+
+"use strict";
+
```
Comment (pcowgill) [inline demo/jest.preprocessor.js]: Should we create an issue and/or add a TODO in the code to remind us of the conditions under which we can remove this file?

File: demo/package.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
     "ios": "npx expo start --ios",
     "eject": "npx expo eject",
     "test": "npx jest",
-    "lint": "npx prettier --write '{*.js,**/*.js,*.jsx,**/*.jsx}' && npx eslint '{*.js,**/*.js,*.jsx,**/*.jsx}'"
+    "lint": "npx prettier --write '{*.js,**/*.js,*.jsx,**/*.jsx}' && npx eslint '{*.js,**/*.js,*.jsx,**/*.jsx,**/*.snap}'"
```
Comment (pcowgill) [inline demo/package.json]: Can you expand on the choice of linting the snapshots? I think it seems reasonable, but I haven't given this much thought before.

File: demo/package.json [hunk 5]
```diff
@@ -23,22 +23,39 @@
...
-    "preset": "jest-expo"
+    "preset": "jest-expo",
+    "transform": {
+      "\\.js$": "<rootDir>/jest.preprocessor.js"
+    },
+    "setupFiles": [
+      "<rootDir>/jest.setup.js"
+    ],
+    "transformIgnorePatterns": [
+      "node_modules/(?!(jest-)?react-native|.*react-navigation|expo-react-native-adapter)"
```
Comment (pcowgill) [inline demo/package.json]: What's the goal of this regex?

File: demo/screens/Home.js [hunk 6]
```diff
@@ -1,16 +1,9 @@
 import React from "react";
 import { Image, Button, StyleSheet, View } from "react-native";
-import { Account } from "tasit-sdk";
 import LargeText from "../components/presentational/LargeText";
 import Colors from "../constants/Colors";
 
 export default class Home extends React.Component {
-  componentDidMount = async () => {
```
Comment (pcowgill) [inline demo/screens/Home.js]: I like that you moved this to the signup component

File: demo/jest.preprocessor.js (hunk 3 above)
Comment (marcelomorgado) [reply demo/jest.preprocessor.js]: Yes, sounds good.

File: demo/package.json (hunk 4 above)
Comment (marcelomorgado) [reply demo/package.json]: I'm not sure if it is a good idea but I saw an eslint rule to keep snapshots small. https://github.com/tasitlabs/tasit/issues/14#issuecomment-448313550
What do you think?

File: demo/jest.preprocessor.js (hunk 3 above)
Comment (pcowgill) [reply demo/jest.preprocessor.js]: Done https://github.com/tasitlabs/tasit/issues/31

File: demo/package.json (hunk 4 above)
Comment (pcowgill) [reply demo/package.json]: Ahhhh got it. I definitely wanted to apply the rule to keep snapshots small - it just didn't cross my mind when reading this line that this change is what would make that possible. Sounds good!

File: demo/package.json (hunk 5 above)
Comment (pcowgill) [reply demo/package.json]: That's a great explanation. Thanks!

**Pull request merged**